CMakeLists.txt

```diff
@@ -281,8 +281,31 @@ add_cppstyle("include-detail" ${CMAKE_CURRENT_SOURCE_DIR}/include/libpmemobj++/d
 add_check_whitespace("include" ${CMAKE_CURRENT_SOURCE_DIR}/include/libpmemobj++/*.hpp)
 add_check_whitespace("include-detail" ${CMAKE_CURRENT_SOURCE_DIR}/include/libpmemobj++/detail/*.hpp)
 
-install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
-	FILES_MATCHING PATTERN "*.hpp")
+if(PKG_CONFIG_FOUND)
+	pkg_check_modules(PMEMOBJ REQUIRED libpmemobj>=1.4)
+else()
+	find_package(PMEMOBJ REQUIRED 1.4)
+endif()
+
+# Check for existence of pmemvlt (introduced after 1.4 release)
+set(SAVED_CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES})
+set(CMAKE_REQUIRED_INCLUDES ${PMEMOBJ_INCLUDE_DIRS})
+CHECK_CXX_SOURCE_COMPILES(
+	"#include <libpmemobj.h>
+	struct pmemvlt vlt;
+	int main() {}"
+	PMEMVLT_PRESENT)
+set(CMAKE_REQUIRED_INCLUDES ${SAVED_CMAKE_REQUIRED_INCLUDES})
+
+if(PMEMVLT_PRESENT)
+	install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+		FILES_MATCHING PATTERN "*.hpp")
+else()
+	message(WARNING "pmemvlt support in libpmemobj not found (to enable - use libpmemobj version > 1.4")
+	install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+		FILES_MATCHING PATTERN "*.hpp"
+		PATTERN "v.hpp" EXCLUDE)
+endif()
```

Comment:
This (two install directives) is going to be painful to update if we will ever support multiple versions of libpmemobj. Is there any better way?

Reply:
Another way (not sure if it's better) would be to have something like this:
install(... FILES_MATCHING PATTERN "*.hpp" PATTERN "v.hpp" EXCLUDE)
if (PMEMVLT_PRESENT)
    install("v.hpp")
endif()

Reply:
Another solution is to *not* use FILES_MATCHING PATTERN and list all files.